hal/stm32f373/spim.c

```diff
@@ -174,10 +197,11 @@ static const struct prescalers_t {
 	{.scale = 256, .reg = SPI_BaudRatePrescaler_256},
 };
 
-void spim_xfer(spim_t *spim, spim_xfer_opts *opts, uint16_t addr, void *read_buf, void *write_buf, int len, spim_xfer_complete complete, void *param)
+bool spim_xfer(spim_t *spim, spim_xfer_opts *opts, uint16_t addr, void *read_buf, void *write_buf, int len, spim_xfer_complete complete, void *param)
```

Comment:
it might be nicer to return an `int` here .... then we can return negative numbers as error codes and positive numbers could be success (ie return `len` is a common api, in case later we need to trim transactions to a max, ie the 65535 limit of the DMA or something)

Reply:
also I think just a general `spi_busy()` function in `spi.c` that `spim.c` and `spis.c` implement as `spim_busy()` and `spis_busy()` would be good too ... it makes the code more explicit

Reply:
hmm thinking a bit more ... maybe the `spim_busy()` is all that we need ... the slave is almost always going to be "busy" really isn't it ... so just adding a flag and a test function to `spim.c` would be good